packages/amplify-appsync-simulator/src/velocity/index.ts

```diff
@@ -112,7 +112,7 @@ export class VelocityTemplate {
       stash: convertToJavaTypes(stash || {}),
       source: convertToJavaTypes(source),
       result: convertToJavaTypes(result),
-      error,
+      error: error ? { ...error, type: error.extensions.errorType } : error,
```

Comment:
I suggest you add a comment block with the reasons for doing this change and linking to the AppSync docs, that way future engineers can understand the context